command_line/cluster_unit_cell.py

```diff
@@ -70,12 +84,81 @@ def run(args=None):
             )
             for expt in experiments
         ]
+    clusters = do_cluster_analysis(crystal_symmetries, params)
+
+    if params.output.clusters:
+        if len(experiments) == 0:
+            print("Clustering output can only be generated for input .expt files")
+            return
+        # Possibilities: either same number of experiments and reflection files,
+        # or just one reflection file containing multiple sequences, or no
+        # reflections given
+        # Want a combined table to work on with experiment identifiers set.
+
+        def _assign_and_return_joint(experiments, reflections):
+            experiments, reflections = assign_unique_identifiers(
+                experiments, reflections
+            )
+            joint_table = flex.reflection_table()
+            for refls in reflections:
+                joint_table.extend(refls)
+            return joint_table
+
+        if len(reflections) == 1:
+            reflections = reflections[0]
+            if len(set(reflections["id"])) != len(experiments):
+                raise ValueError(
+                    f"Mismatched number of reflection tables (f{len(set(reflections['id']))}) and experiments (f{len(experiments)})"
+                )
+            if not dict(reflections.experiment_identifiers()):
+                reflections = reflections.split_by_experiment_id()
+                reflections = _assign_and_return_joint(experiments, reflections)
+
+        elif len(reflections) > 1:
+            if not len(reflections) == len(experiments):
+                reflections = parse_multiple_datasets(reflections)
+                if len(reflections) != len(experiments):
+                    raise ValueError(
+                        f"Mismatched number of reflection tables (f{len(reflections)}) and experiments (f{len(experiments)})"
+                    )
+            reflections = _assign_and_return_joint(experiments, reflections)
```

Comment:
This seems like something that is probably used in other command line (or very similar code) - could it be extracted somewhere so it can be more easily re-used? (not necessarily as part of this PR)

Reply:
Sure, will keep this in mind when reviewing the other relevant tools for multicrystal work